continuum/scenarios/scenario_utils.py

```diff
@@ -4,39 +4,123 @@
 from torch.utils.data import DataLoader
 
 from continuum.datasets import InMemoryDataset, H5Dataset
-from continuum.scenarios import ContinualScenario
+from continuum.scenarios import ContinualScenario, OnlineFellowship
 from continuum.tasks import TaskType
 
 
+def _get_remapping_classes_ascending_order(new_classes, current_mapping=None):
+    """
+    Output a vector of classes existing class to get new class labels do:
+     new_label = np.where(new_remapping==old_label)[0][0]
+
+    :param new_classes: list of new classes
+    :param current_mapping: vector with previous remapping
+    """
+
+    array_new_classes = np.array(new_classes)
+
+    if len(np.unique(array_new_classes)) != len(array_new_classes):
+        raise ValueError("list new_classes can not contain two time the same class label.")
+
+    ordered_array_new_classes = np.sort(array_new_classes)
+
+    if current_mapping is None:
+        new_remapping = ordered_array_new_classes
+    else:
+        # remove classes already in the mapping
+        array_new_classes = np.setdiff1d(array_new_classes, current_mapping)
+        if len(array_new_classes) == 0:
+            new_remapping = current_mapping
+        else:
+            new_remapping = np.concatenate([current_mapping, array_new_classes], axis=0)
+
+    return new_remapping
+
+def _remap_class_vector(class_vector, remapping):
+
+    if len(np.where(class_vector == -1)[0]) > 0:
+        raise ValueError("-1 is not an acceptable label.")
+
+    if len(np.setdiff1d(class_vector, remapping)) > 0:
+        raise ValueError("Some values in class vector are not in the mapping.")
+
+    # we create a new vector to not have interference between old classes and new classes values
+    new_vector = np.ones(len(class_vector)) * -1
+    for i, key in enumerate(remapping):
+        indexes = np.where(class_vector == key)[0]
+        new_vector[indexes] = i
+
+    if len(np.where(new_vector == -1)[0]) > 0:
+        raise ValueError("Some indexes have not been set in the remapping.")
+    return new_vector
+
+def remap_class_vector(class_vector, remapping=None):
```

Comment:
Isn't it the mapping already done in https://github.com/Continvvm/continuum/blob/master/continuum/scenarios/class_incremental.py#L69 ? (much simpler and probably faster), or am i missing something?